### PR TITLE
Added grades app

### LINKED ISSUE #1957: Final grades: create model
Part of #1856 

In order to have persistent, immutable source of truth for final grades, we need a model -- as opposed to continuously updating grades from edX.org

### Fields:

- user
- course run (course key)
- grade (between 0 and 1)
- pass/fail, 
- status (pending | complete)

### Acceptance criteria:

- [ ] includes an audit model similar to financial aid

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
closes #1957 

#### What's this PR do?
Adds app and models for final grades. The final grades model has an audit table associated.

#### Where should the reviewer start?
`grades/models.py`

#### How should this be manually tested?
Test that you can associate only a final grade between 0 and 1 to an user for a course.

#### Any background context you want to provide?
I did not create a constraint to avoid the deletion of the audit entries yet.

### CODE REVIEW

**Comment by noisecapella** [comment]
> I did not create a constraint to avoid the deletion of the audit entries yet.
That's fine, can you make an issue for this?

File: grades/models.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+class FinalGrade(TimestampedModel, AuditableModel):
+    """
+    Model to store final grades
+    """
+    user = models.ForeignKey(User, null=False)
+    course_run = models.ForeignKey(CourseRun, null=False)
+    grade = models.FloatField(
+        null=True,
+        validators=[MinValueValidator(0.0), MaxValueValidator(1.0)]
```
Comment (noisecapella) [inline grades/models.py]: When do these validators get run?

**Comment by noisecapella** [comment]
Just some questions, overall things look good

**Comment by giocalitri** [comment]
>> I did not create a constraint to avoid the deletion of the audit entries yet.

>That's fine, can you make an issue for this?

done #2000

File: grades/admin.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Admin for the grades app
+"""
+from django.contrib import admin
+
+from grades import models
+from micromasters.utils import get_field_names
+
+
+class FinalGradeAdmin(admin.ModelAdmin):
+    """Admin for FinancialAid"""
```
Comment (giocalitri) [inline grades/admin.py]: sure

File: grades/models.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+from courses.models import CourseRun
+from micromasters.models import (
+    AuditableModel,
+    AuditModel,
+    TimestampedModel,
+)
+from micromasters.utils import serialize_model_object
+
+
+class FinalGradeStatus:
```
Comment (giocalitri) [inline grades/models.py]: I am not sure if we need just these 2 statuses or we will end up having more...

**Comment by noisecapella** [comment]
Functionality looks good. What did you want to do about validation? I feel like validating it in `save()` is the only option if we want to do it at the model level

**Comment by giocalitri** [comment]
I actually pushed some code to do the validation in the `.save()` method

**Comment by noisecapella** [comment]
Oh right, I missed the `full_clean` :+1:

**Comment by noisecapella** [comment]
Actually can you add a test that validation happens when `save` is called?

**Comment by giocalitri** [comment]
Sure

**Comment by giocalitri** [comment]
@noisecapella I added the test

File: grades/admin.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        return False
+
+    def save_model(self, request, obj, form, change):
+        """
+        Saves object and logs change to object
+        """
+        obj.save_and_log(request.user)
+
+
+class FinalGradeAdminAuditAdmin(admin.ModelAdmin):
```
Comment (noisecapella) [inline grades/admin.py]: Can we rename this to `FinalGradeAuditAdmin`, just to remove the redundant `Admin`?

**Comment by noisecapella** [comment]
Looks great, just some nitpicking :+1:

File: grades/admin.py (hunk 4 above)
Comment (giocalitri) [inline grades/admin.py]: yep

**Comment by giocalitri** [comment]
done: will merge as soon as tests pass

**Pull request merged**